packages/sil-governance/src/routes/create-proposal/index.tsx

```diff
@@ -14,10 +15,12 @@ import ProposalForm from "./components/ProposalForm";
 const CreateProposal = (): JSX.Element => {
   const dispatch = useDispatch();
   const lastSignAndPostResult = useSelector((state: RootState) => state.transactions.lastSignAndPostResult);
+  const proposals = useSelector((state: RootState) => state.proposals);
 
   const onReturnToDashboard = (): void => {
     dispatch(setTransactionsStateAction());
     history.push(DASHBOARD_ROUTE);
+    dispatch(replaceProposalsAction(proposals));
```

Comment:
Why this change? Here you take data from the store and write it to the store.

To reload from chain, we need something like

```ts
  const loadProposals = async (): Promise<void> => {
    const governor = store.getState().extension.governor;
    if (!governor) throw new Error("Governor not set in store. This is a bug.");
    const chainProposals = await getProposals(governor);
    dispatch(replaceProposalsAction(chainProposals));
  };
```

Reply:
Sure, I messed up there because I began to follow the suggested types... The issue I'm having is `governor.getProposals()` returns `Proposal[]` and `replaceProposalsAction` expects `ProposalProps[]`.

Reply:
When you get the transaction ID, it was just posted into the network. I.e. it is not yet included in a block. This is the reason I suggetsed a 5 second delay between getting the result and triggering a reload

Reply:
The delay should be before showing the confirmation page, or before redirecting to Dashboard after confirming the return?

Reply:
The best thing would be to have the reload independent of the screen navigation, like 5 seconds after you received the transaction ID (triggered by `setTimeout`)